tools/revng-lift/CodeGenerator.cpp

```diff
@@ -1268,6 +1268,34 @@ void CodeGenerator::translate(Optional<uint64_t> RawVirtualAddress) {
 
   Variables.setDataLayout(&TheModule->getDataLayout());
 
+  // We already finished discovering new code to translate, so we can remove
+  // the references to local variables as argument of the calls to newpc and
+  // create room for more optimizations.
+  auto *NewPC = TheModule->getFunction("newpc");
```

Comment:
You mean to move the whole change inside `finalizeNewPCMarkers`, possibly after the first if-condition, right?

Reply:
Yeah in the `if`

Reply:
Shouldn't it have been called before running SROA (line 1302)?

Reply:
Makes sense. Looks fine to me to anticipate `finalizeNewPCMarkers`.

I'm only in doubt what the following piece of code was for:

```c++
      // TODO: by default we should leave these                                                                                                                                               
      for (unsigned I = 4; I < ArgCount - 1; I++)
        Call->setArgOperand(I, Call->getArgOperand(ArgCount - 1));
```

Reply:
Seems like it's replacing the second to last pointer (to local vars) with the value of the very last one. Considered that we are removing these pointers, this could be dropped, couldn't it?

Reply:
Did some tests, so I decided to take some liberties and simplify the whole code. Hope it's OK.

Reply:
Apparently some tests fail when moving `finalizeNewPCMarkers` before the PassManager; [this assert](https://github.com/revng/revng/blob/develop/tools/revng-lift/JumpTargetManager.h#L350) is triggered. Investigating this.

Reply:
Stack trace? Which pass?

Reply:
Tried to anticipate `createJTReasonMD` before the pipeline too, but doesn't work. Stack trace [here](https://ghostbin.co/paste/pkyfu).

Reply:
You moved `finalizeJumpTargets` instead of `finalizeNewPCMarkers`.

Reply:
Oh man, the inconveniences of using another machine to run the test suite and having two instances of VSCode opened  of the same code (must have happened when moving a loc from one to another for tests), thanks.